src/main/java/me/crespel/runtastic/RuntasticExportConverter.java

```diff
@@ -34,73 +42,440 @@ public static void main(String[] args) {
+				System.out.println("      Created At: " + sdf.format(user.getCreatedAt()) + ",  Confirmed At: " + sdf.format(user.getConfirmedAt()) + ",  Last Sign-in At: " + sdf.format(user.getLastSignInAt()) + ",  Updated At: " + sdf.format(user.getUpdatedAt()));
+			}
+		}
+	}
+
+	protected void doPhoto(File path, String id) throws FileNotFoundException, IOException {
+		SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
+		SportSession session = converter.getSportSessionWithPhoto(path, id);
+		for (ImagesMetaData image : session.getImages()) {
+			if (image != null) {
+				if (image.getId() == Integer.parseInt(id)) {
+					doInfo(path, session.getId());
+					System.out.println(sdf.format(session.getStartTime()) + " - ID: " + session.getId());
+					System.out.println("             [" + image.getId() + ".jpg] " + sdf.format(image.getCreatedAt()) + ": " + image.getDescription() );
+					if( image.getLatitude() != null ) System.out.println("             ( http://maps.google.com/maps?q=" + image.getLatitude() + "," + image.getLongitude() + " )");
+				}
+			}
 		}
 	}
 
 	protected void doConvert(File path, String id, File dest, String format) throws FileNotFoundException, IOException {
 		if ("all".equalsIgnoreCase(id)) {
 			long startTime = System.currentTimeMillis();
-			int count = converter.convertSportSessions(path, dest, format);
+			int count = converter.exportSportSessions(path, dest, format);
 			long endTime = System.currentTimeMillis();
 			System.out.println(count + " activities successfully written to '" + dest + "' in " + (endTime - startTime) / 1000 + " seconds");
 		} else {
-			converter.convertSportSession(path, id, dest, format);
+			converter.exportSportSession(path, id, dest, format);
 			System.out.println("Activity successfully written to '" + dest + "'");
 		}
 	}
 
+	private void doOverlap(File path, File dest, String format) throws FileNotFoundException, IOException {
+		long startTime = System.currentTimeMillis();
+		System.out.println("Load full list of sport session (inclusive all sub-data), this requires some time ...");
+		List<SportSession> sessions = converter.convertSportSessions(path, format);
+		doOverlap(sessions);
+		displayOverlapSummary(sessions, false);
+
+		for( SportSession session : sessions) {
+			List<SportSession> overlapsessions = session.getOverlapSessions();
+			if( (dest!=null) && (overlapsessions!=null) && (overlapsessions.size() > 0) ) {
+				converter.exportSportSession(session, dest, format);
+			}
+		}
+
+		long endTime = System.currentTimeMillis();
+		System.out.println(sessions.size() + " activities successfully processed, in " + (endTime - startTime) / 1000 + " seconds");
+	}
+
+	// Loop through all sport session and add "overlapping" session to each sport session
+	private void doOverlap(List<SportSession> sessions) {

```

Comment:
Overlap processing (i.e. with no user interaction) should be moved to `ExportConverter` instead.

Displaying information about overlapping sessions can stay in this class however.

Reply:
Thanks, moved "process" methods to export converter class.